solutions/balkan-11-timeismoney.mdx

```diff
@@ -0,0 +1,186 @@
+---
+id: balkan-11-timeismoney
+source: Balkan OI 2011
+title: Timeismoney
+author: Chuyang Wang
+---
+
+[Official Analysis](http://www.boi2011.ro/resurse/tasks/timeismoney-sol.pdf)
+
+## Explanation
+
+For the 40% $t = c$ cases, we can solve the problem by constructing an MST with e.g. $t$ as the weights for the edges. Let $T = \sum t$ and $C = \sum c$. $V = T \cdot C = T^2$ will be minimized when $T$ is minimized. 
+
+To solve the problem completely, we will have to minimize the product of those two sums. The solution space contains all possible MSTs with their respective $T$ and $C$ values. We can view each of those possible solutions as a 2D-point $(T, C)$. The inverse proportion function $C = \frac{V}{T}$ describes the hyperbola on which all the points have the same $V$ value. This hyperbola is convex, so it suffices to check all points on the lower convex hull of our solution space to find the minimal $V$ value. Nevertheless, we cannot really calculate the convex hull since it would require all $\binom{M}{N-1}$ points to be known. Instead, we use another approach below. 
```

Comment:
> The inverse proportion function $C = \frac{V}{T}$ describes the hyperbola on which all the points have the same $V$ value.

what

Reply:
V = C \cdot T => C = \frac{V}{T}
V is defined in the problem statement so I didn't mention it. Should I add a sentence defining V?

Reply:
ok but, hyperbola?

Reply:
maybe i should use "curve"? (i mean it _is_ the positiv part of the hyperbola)